File: Тренировка 1.0/02. Линейный поиск/A. Возрастает ли список/Возрастает ли список.cpp
#include <iostream>

using namespace std;

int main()
{
    char oneElement;
    int curNumber;
    cin >> curNumber;
    cin.get(oneElement);
 
    bool nonDecreasing = true;
    if (oneElement != '\n')
    {
        do
        {
            int newNumber;
            cin >> newNumber;
            if (newNumber <= curNumber)
            {
                nonDecreasing = false;
                break;
            }
            curNumber = newNumber;
        } while (cin.get() != '\n');
    }
    cout << (nonDecreasing ? "YES" : "NO");
}

File: Тренировка 1.0/02. Линейный поиск/A. Возрастает ли список/Возрастает ли список.cs
using System;

string[] list = Console.ReadLine().Split(' ');

if (list.Length == 1) Console.WriteLine("YES");
else
{
    bool nonDecreasing = true;
    for(int i = 1; i < list.Length; i++)
    {
        if (int.Parse(list[i]) <= int.Parse(list[i - 1]))
        {
            nonDecreasing = false;
            break;
        }
    }
    Console.WriteLine(nonDecreasing ? "YES" : "NO");
}